service-app/internal/httpclient/request.go

```diff
@@ -53,24 +60,34 @@ func (r *HttpClient) HTTPRequest(ctx context.Context, url, method string, payloa
 	}
 	defer resp.Body.Close() //nolint:errcheck // no need to check error when closing body
 
+	body, err := io.ReadAll(resp.Body)
+	if err != nil {
+		return nil, fmt.Errorf("failed to read response body: %w", err)
+	}
+
 	// Handle non-2xx responses
-	if resp.StatusCode == 404 {
-		return nil, ErrNotFound
+	if resp.StatusCode >= 400 && resp.StatusCode < 500 {
+		errOut := SiriusClientError{
+			StatusCode: resp.StatusCode,
+		}
+
+		if resp.StatusCode == 400 {
+			var respBody struct {
+				ValidationErrors map[string]map[string]string `json:"validation_errors"`
+			}
+
+			if json.Unmarshal(body, &respBody) == nil {
+				errOut.ValidationErrors = respBody.ValidationErrors
```

Comment:
unless you are expecting to extend this type for other cases I think it would be neater to do it like `ErrNotFound` before, so have the check for `caseReference` here and return an `ErrInvalidUID`

Reply:
I started with that, but it felt wrong because this is a relatively generic HTTP client that's also used for other requests to Sirius. So I tried to separate out the logic of HTTP transfer errors from the logic of the case number being checked.